### ISSUE TITLE
NewMessage event not working for majority of channels all of a sudden

### ISSUE THREAD

**Issue by himanshuj89** [gram-js/gramjs#682]
Hi,

I am subscribed to 20+ Telegram channels and I was receiving all new messages until 2 days back without any issues. 

Since Yesterday, I am not receiving message for NewMessage event for most of the channels. I didn't make any changes in the code. 

I am only receiving this for 1-2 channels and not others. On Telegram app, I can see messages on almost all channels. 

I am using below line of code for subscribing to all new messages - 

`client.addEventHandler(eventFunction, new NewMessage({}));`

Any idea if Telegram updated something at their end? 

thanks.

**Comment by boltpl81** [comment]
I thought that I messed something in the code, but it looks like it's genereral problem

**Comment by himanshuj89** [comment]
Yes, correct. In my case, I am getting messages through event for channels with subscriber count around 1000 but not for channels with subscriber count more than 30000.

**Comment by boltpl81** [comment]
Couple of minutes ago I've just received event from huge channel. Maybe some hickup on telegram end? I'm going to monitor the behaviour.

**Comment by GuilhermeGabriel** [comment]
Same here

**Comment by justalike** [comment]
can confirm, this happens to me as well.
first occurrence was 5 days ago.

feels like something internal has changed in telegram core.

**Comment by GuilhermeGabriel** [comment]
> can confirm, this happened to me as well.
> 
> So I've noticed that connection was established using Layer 159 which is kinda outdated?
> 
> So what I've done: Tried to do hard reinstall of the library by
> 
> * removing it with `npm remove telegram` and
> * reinstalling back with `npm install telegram`
> 
> Now logs say Layer 179 is used for connecting. All of a sudden got a message from the channel with 25k+ subscribers.
> 
> Tested out (not too extensively though)
> 
> So update your lib to ` "telegram": "^2.21.2"` to test it out.
> 
> Fixed the issue for me with no code changes.

This fixed the problem here. Thank you!

**Comment by SwatX18** [comment]
> can confirm, this happened to me as well.
> 
> So I've noticed that connection was established using Layer 159 which is kinda outdated?
> 
> So what I've done: Tried to do hard reinstall of the library by
> 
> * removing it with `npm remove telegram` and
> * reinstalling back with `npm install telegram`
> 
> Now logs say Layer 179 is used for connecting. All of a sudden got a message from the channel with 25k+ subscribers.
> 
> Tested out (not too extensively though)
> 
> So update your lib to ` "telegram": "^2.21.2"` to test it out.
> 
> Fixed the issue for me with no code changes.

Still not working for me...

**Comment by nickuya** [comment]
I was running into the same issue. When I started up my program, I would receive new messages for my large channels but after about 2 minutes it would stop.

One day I left the program running anyways, and then after about 10 minutes I noticed a timeout error in my logs and started receiving new messages again.  I tracked down the timeout to here
https://github.com/gram-js/gramjs/blob/5a0f45b2b5f30472634d9948cbc404b09b54e0c7/gramjs/client/updates.ts#L245-L260

so now, on an interval, I'm trying to just call the same reconnect method as at the bottom of that error handler.

```
      setInterval(async () => {
        client._sender?.reconnect();
      }, 1000 * 60);
```

seems to work for me so far. Hope it works for others.

**Comment by sbeben** [comment]
same here
also, the workaround proposed by @nickuya does not work as reconnects are never successful for some reason

**Comment by MishaMgla** [comment]
Experiencing same issue. With some messages it works flawlessly, with some NewMessage event is never triggered.

**Comment by egarciahz** [comment]
I'v been experiencing this same issue the last two days, did anyone managed to fix it ?

**Comment by thegauravparmar** [comment]
Any update on this? facing same, suddenly stopped receiving events

**Comment by indapublic** [comment]
Same

**Comment by firdiar** [comment]
bump, same here

**Comment by fzn0x** [comment]
Any updates?

**Comment by abyrwalg** [comment]
I started having this issue today.

**Comment by fzn0x** [comment]
> I started having this issue today.

Same here, only messages from session owner being captured.

**Comment by joaoabvs** [comment]
Same here, only messages from session owner being captured.

**Comment by StepanyanAlbert** [comment]
Same thing here 
All of a sudden messages stopped coming.
I have channel with 9 subs , works pretty fine, the rest are gone, no message , I tried using Raw instead of NewMessage but no luck at all

**Comment by tkddlf4209** [comment]
You can use "mtcute" as an alternative to "gramjs". I checked today's update and confirmed that it works properly even on large channels.

https://github.com/mtcute/mtcute

**Comment by sliterok** [comment]
I had the same issue, I tried to migrate to mtcute per advice from @tkddlf4209 and I encountered the same problem. From my home I can see events for all channels, but on VDS the events from some channels disappear.
However, if you open channel by passing its id to this method https://mtcute.dev/guide/intro/updates.html#opening-chats it works! Maybe it would be possible to implement such method in gramjs? It seems to be calling [notifyChannelOpened](https://github.com/mtcute/mtcute/blob/57ab769a76db94fc5ad60cad12f3dd53583487a2/packages/core/src/highlevel/methods/chats/open-chat.ts#L32C18-L32C37) method in client

**Comment by fzn0x** [comment]
Guys if you still encountered this issue, unfortunately the solution is to change the library or build your own from scratch.

I'm using https://www.npmjs.com/package/tdl right now and it's working fine.

Gramjs seems outdated to handle cases like this.

To save your time just ask claude AI to migrate your code to `tdl`

**Comment by joaoabvs** [comment]
> Guys if you still encountered this issue, unfortunately the solution is to change the library or build your own from scratch.
> 
> I'm using https://www.npmjs.com/package/tdl right now and it's working fine.
> 
> Gramjs seems outdated to handle cases like this.
> 
> To save your time just ask claude AI to migrate your code to `tdl`

How has the latency been when receiving new messages compared to GramJS? Do updates arrive instantly or have you noticed any delays?

**Comment by fzn0x** [comment]
> > Guys if you still encountered this issue, unfortunately the solution is to change the library or build your own from scratch.
> > I'm using https://www.npmjs.com/package/tdl right now and it's working fine.
> > Gramjs seems outdated to handle cases like this.
> > To save your time just ask claude AI to migrate your code to `tdl`
> 
> How has the latency been when receiving new messages compared to GramJS? Do updates arrive instantly or have you noticed any delays?

Updates arrive instantly on my end

**Comment by StepanyanAlbert** [comment]
> > > Guys if you still encountered this issue, unfortunately the solution is to change the library or build your own from scratch.
> > > I'm using https://www.npmjs.com/package/tdl right now and it's working fine.
> > > Gramjs seems outdated to handle cases like this.
> > > To save your time just ask claude AI to migrate your code to `tdl`
> > 
> > 
> > How has the latency been when receiving new messages compared to GramJS? Do updates arrive instantly or have you noticed any delays?
> 
> Updates arrive instantly on my end

Switch to TDL , works pretty fine. There are couple of good Node js wrappers if you do not want to work with tdl in raw json format
I use https://github.com/eilvelia/tdl

**Comment by telegreen** [comment]
Guys! 
I've used the Teleproto library forked from GramJS and fully compatible. 
Everything works perfectly without any code changes.

teleproto 1.221.0 Updated to MTProto Layer 221

https://www.npmjs.com/package/teleproto/v/1.221.0


Just made minor change to package.json and my project is working again. 

```
"telegram": "npm:teleproto@1.221.0"
```

**Comment by StepanyanAlbert** [comment]
> Guys! I've used the Teleproto library forked from GramJS and fully compatible. Everything works perfectly without any code changes.
> 
> teleproto 1.221.0 Updated to MTProto Layer 221
> 
> https://www.npmjs.com/package/teleproto/v/1.221.0
> 
> Just made minor change to package.json and my project is working again.
> 
> ```
> "telegram": "npm:teleproto@1.221.0"
> ```

Works for me as well (just for test, I already switched to TDL)

**Comment by fzn0x** [comment]
> Guys! I've used the Teleproto library forked from GramJS and fully compatible. Everything works perfectly without any code changes.
> 
> teleproto 1.221.0 Updated to MTProto Layer 221
> 
> https://www.npmjs.com/package/teleproto/v/1.221.0
> 
> Just made minor change to package.json and my project is working again.
> 
> ```
> "telegram": "npm:teleproto@1.221.0"
> ```

Banger

**Comment by sbeben** [comment]
Nah, seems like both mtqute and teleproto are randomly skipping updates as well

**Comment by toorhamza** [comment]
I switched to teleproto first but that didn't work but after switching to [](https://www.npmjs.com/package/td), it's working now.

The channel I wanted to get events for new messages had 123,000+ subscribers

**Comment by marcuth** [comment]
Using the [`teleproto`](https://www.npmjs.com/package/teleproto) solved the problem, but only partially. I noticed that some messages are still being lost.
